src/Libraries/Microsoft.Extensions.AI.Evaluation.NLP/Utilities/CollectionBuilderAttribute.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if !NET8_0_OR_GREATER
```

Comment:
Can this be moved under the `src\LegacySupport` folder with an accompanying `InjectCollectionBuilderOnLegacy` property in [LegacySupport.props](https://github.com/dotnet/extensions/blob/main/eng/MSBuild/LegacySupport.props)?